src/SprykerEco/Zed/Unzer/Business/Quote/UnzerQuoteExpander.php

```diff
@@ -0,0 +1,182 @@
+
+class UnzerQuoteExpander implements UnzerQuoteExpanderInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerCustomerAdapterInterface
+     */
+    protected $customerAdapter;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Quote\Mapper\UnzerQuoteExpanderMapperInterface
+     */
+    protected $unzerQuoteExpanderMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Dependency\UnzerToQuoteClientInterface
+     */
+    protected $quoteClient;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\UnzerConfig
+     */
+    protected $unzerConfig;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface
+     */
+    protected $unzerReader;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\ApiAdapter\UnzerCustomerAdapterInterface $customerAdapter
+     * @param \SprykerEco\Zed\Unzer\Business\Quote\Mapper\UnzerQuoteExpanderMapperInterface $unzerQuoteExpanderMapper
+     * @param \SprykerEco\Zed\Unzer\Dependency\UnzerToQuoteClientInterface $quoteClient
+     * @param \SprykerEco\Zed\Unzer\UnzerConfig $unzerConfig
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface $unzerReader
+     */
+    public function __construct(
+        UnzerCustomerAdapterInterface $customerAdapter,
+        UnzerQuoteExpanderMapperInterface $unzerQuoteExpanderMapper,
+        UnzerToQuoteClientInterface $quoteClient,
+        UnzerConfig $unzerConfig,
+        UnzerReaderInterface $unzerReader
+    ) {
+        $this->customerAdapter = $customerAdapter;
+        $this->unzerQuoteExpanderMapper = $unzerQuoteExpanderMapper;
+        $this->quoteClient = $quoteClient;
+        $this->unzerConfig = $unzerConfig;
+        $this->unzerReader = $unzerReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function expand(QuoteTransfer $quoteTransfer): QuoteTransfer
+    {
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== SharedUnzerConfig::PROVIDER_NAME) {
+            return $quoteTransfer;
+        }
+
+        $quoteTransfer = $this->createUnzerPayment($quoteTransfer);
+        $quoteTransfer = $this->createUnzerCustomer($quoteTransfer);
+        if ($quoteTransfer->getPayment()->getUnzerPayment()->getIsMarketplace()) {
+            $quoteTransfer = $this->expandQuoteItemsWithUnzerParticipants($quoteTransfer);
+        }
+
+        $this->quoteClient->setQuote($quoteTransfer);
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    protected function createUnzerCustomer(QuoteTransfer $quoteTransfer): QuoteTransfer
+    {
+        if (
+            $quoteTransfer->getPayment()->getUnzerPayment() !== null &&
+            $quoteTransfer->getPayment()->getUnzerPayment()->getCustomer() !== null
```

Comment:
Imho private helper method would increase readability

Reply:
But it will lead to one more method with single usage